stable/database/files/nuosm

```diff
@@ -276,235 +405,312 @@ else
+          # restore_source_encoded variable should be passed in the user_data by
+          # `nuorestore` script
+          die 1 "restore source is missing from restore request for archiveId=${myArchive}, userData='${user_data}'"
+        fi
+        restore_source="$(printf "%s" "${restore_source_encoded}" | base64 -d)"
+        resolveLatestSource || die 1 "unable to resolve ${restore_source} restore source"
+        # restore chart doesn't have notion of stream|backupset
+        # `nuodocker restore archive` works with both
+        restore_type="backupset"
+        [ -z "$restore_credentials_encoded" ] && restore_credentials_encoded="$(printf "%s" "${DATABASE_RESTORE_CREDENTIALS:-:}" | base64)"
+        [ -z "$strip_levels" ] && strip_levels=${DATABASE_RESTORE_STRIP_LEVELS:-1}
+        
+        log "Archive restore will be performed for archiveId=${myArchive}, source=${restore_source}, type=${restore_type}, strip=${strip_levels}"
+        trace "performing restore for archiveId=${myArchive}"
+        if isRestoreSourceAvailable "$restore_source"; then
+          perform_restore || die $? "$error"
+        else
+          error="Backupset $restore_source cannot be found in $NUODB_BACKUPDIR"
+          die 1 "$error"
+        fi
+      fi
+    elif [ "$atomCount" -lt 20 ] && [ "$catalogCount" -lt 2 ]; then
+      if [ -n "$NUODB_AUTO_RESTORE" ]; then
+        # autoRestore is configured, try to REPAIR the disk archive
+        restore_source="$NUODB_AUTO_RESTORE"
+        resolveLatestSource || log "unable to resolve ${restore_source} restore source"
+        restore_credentials_encoded="$(printf "%s" "${DATABASE_RESTORE_CREDENTIALS:-:}" | base64)"
+        restore_type="${NUODB_AUTO_RESTORE_TYPE}"
+        strip_levels="${NUODB_RESTORE_STRIP_LEVELS:-1}"
+
+        log "Automatic archive repair will be performed for archiveId=${myArchive}, source=${restore_source}, type=${restore_type}, strip=${strip_levels}"
+        trace "restoring damaged archive"
+        if isRestoreSourceAvailable "$restore_source"; then
+          perform_restore || die $? "$error"
+        fi
+        # delete damaged archive metadata as a new one will be created by
+        # nuodocker start sm
+        [ ! -f "$DB_DIR/info.json" ] && purgeArchive "$myArchive"
+      fi
+    fi
+  elif [ ! -f "$DB_DIR/1.atm" ]; then
+    if [ -n "$NUODB_AUTO_IMPORT" ] && isRestoreSourceAvailable "$NUODB_AUTO_IMPORT"; then
+      # autoImport is configured, try to IMPORT the disk archive
+      restore_source="$NUODB_AUTO_IMPORT"
+      restore_credentials_encoded="$(printf "%s" "${DATABASE_IMPORT_CREDENTIALS:-:}" | base64)"
+      restore_type=${NUODB_AUTO_IMPORT_TYPE}
+      strip_levels=${NUODB_IMPORT_STRIP_LEVELS:-1}
+
+      log "Automatic archive import will be performed for archiveId=${myArchive}, source=${restore_source}, type=${restore_type}, strip=${strip_levels}"
+      trace "restoring empty archive"
+      perform_restore || die $? "$error"
+    fi
+  fi
+else
```

Comment:
The code that should be reviewed ends here.
The rest of the code is just indented with minor changes.